### ISSUE TITLE
Incorrect error message with strtr function

### ISSUE THREAD

**Issue by Riimu** [vimeo/psalm#831]
The following piece of code: https://getpsalm.org/r/1f80193969

```php
<?php

strtr(file_get_contents('foobar.txt'), [
    'foo' => 'bar',
]);
```

Produces the error:

```
ERROR: InvalidArgument - 3:40 - Argument 2 of strtr expects string, array{foo:string(bar)} provided
```

The actual error should be that the the first argument must have a type `string`, but `file_get_contents()` returns `string|false`. It correctly identifies that there is no matching signature, but I'm guessing due to multiple definitions of `strtr` in the call map, the actual error message ends up being misleading.

**Comment by muglug** [comment]
There shouldn’t be any issue emitted here, because Psalm doesn’t worry about the possibly-false return value of `file_get_contents`. And you’re right about the cause, too.

**Comment by muglug** [comment]
Thanks!
